### PR TITLE
use package:test to run tests

### PR DESCRIPTION
**Pull request by devoncarew**
- switch to using package:test to run tests. This is largely a mechanical change, w/ some work to get the pubspec right - this will get simpler if and when we drop the dep on 'package:analyzer`
- fix https://github.com/dart-lang/dart-services/issues/256

@danrubel @lukechurch

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -22,15 +22,20 @@ dependencies:
...
-  unittest: ^0.11.0
+  test: ^0.12.0
 
 executables:
   services: null
+
+dependency_overrides:
+  # package:test's version constraint does not include analyzer 0.30.0 alpha
+  analyzer: 0.30.0-alpha.3
+  # package:shelf_route's version constraint doesn't not include matcher 0.12.0
```
Comment (lukechurch) [inline pubspec.yaml]: nit 'doesn't not' -> 'does not'?

**Comment by lukechurch** [comment]
LGTM with nit

When we're done with this, lets also chat about how we might integrate the longer tests that we run as a pre-deployment script into this.

File: pubspec.yaml (hunk 1 above)
Comment (devoncarew) [inline pubspec.yaml]: oops

**Comment by devoncarew** [comment]
Yeah, there was some legerdemain around `gae_deployed_test.dart` - to keep the spirit of the `unittest` based tests - but definitely something we can improve here.

**Pull request merged**